e2e/assert.go

```diff
@@ -199,9 +201,16 @@ func fetchStatefulSet(sb testenv.DiffingSandbox, name string) (*appsv1.StatefulS
 }
 
 func fetchPodsInStatefulSet(sb testenv.DiffingSandbox, labels map[string]string) ([]corev1.Pod, error) {
-	var pods corev1.PodList
-
-	if err := sb.List(&pods, labels); err != nil {
+	matchingLabels := client.MatchingLabels(labels)
+	// create a new clientset to talk to k8s
+	clientset, err := kubernetes.NewForConfig(sb.Mgr.GetConfig())
+	if err != nil {
+		return nil, err
+	}
+	pods, err := clientset.CoreV1().Pods(sb.Namespace).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: k8slabels.Set(matchingLabels).AsSelector().String(),
+	})
+	if err != nil {
```

Comment:
Should we check on the different error types from k8s?

Reply:
let's leave it generic for now